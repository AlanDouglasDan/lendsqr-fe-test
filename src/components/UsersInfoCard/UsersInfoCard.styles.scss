.grid-card {
    // height: 160px;
    background-color: #fff;
    padding: 20px 30px;
    border: 1px solid rgba(33, 63, 125, 0.06);
    box-shadow: 3px 5px 20px rgba(0, 0, 0, 0.04);
    border-radius: 4px;
  }
  
  .card-icon {
    width: 40px;
    height: 40px;
    border-radius: 50%;
  }
  
  .card-text {
    font-size: 16px;
    color: #545f7d;
    margin: 12px 0;
    font-weight: 500;
    text-transform: uppercase;
  }
  
  .card-text-bold {
    font-size: 24px;
    color: #213f7d;
    font-weight: 600;
  }
  
  .light-pink {
    background-color: rgba(223, 24, 255, 0.1);
  }
  
  .light-blue {
    background-color: rgba(87, 24, 255, 0.1);
  }
  
  .light-orange {
    background-color: rgba(245, 95, 68, 0.1);
  }
  
  .light-red {
    background-color: rgba(255, 51, 102, 0.1);
  }
  