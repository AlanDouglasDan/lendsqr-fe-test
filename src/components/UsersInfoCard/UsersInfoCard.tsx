import "./UsersInfoCard.styles.scss";

type Props = {
  icon: JSX.Element;
  color: string;
  title: string;
  number: string;
};

const UsersInfoCard: React.FC<Props> = ({ icon, color, title, number }) => {
  return (
    <div className="grid-card">
      <div
        className={`card-icon align-items-center justify-content-center d-flex ${color}`}
      >
        {icon}
      </div>

      <div className="card-text">{title}</div>
      <div className="card-text-bold">{number}</div>
    </div>
  );
};

export default UsersInfoCard;
