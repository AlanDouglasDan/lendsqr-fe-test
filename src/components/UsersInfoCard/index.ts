export { default as UsersInfoCard } from './UsersInfoCard';
