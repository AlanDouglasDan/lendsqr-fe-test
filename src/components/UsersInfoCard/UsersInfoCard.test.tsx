import { BrowserRouter } from "react-router-dom";
import { render } from "@testing-library/react";

import UsersInfoCard from "./UsersInfoCard";

describe("Users Info Card Component Test", () => {
  it("Users Info Card Component Snapshot", () => {
    const { asFragment } = render(
      <BrowserRouter>
        <UsersInfoCard
          icon={<div />}
          title="hello world"
          number="hello world"
          color="blue"
        />
      </BrowserRouter>
    );
    expect(asFragment()).toMatchSnapshot();
  });
});
