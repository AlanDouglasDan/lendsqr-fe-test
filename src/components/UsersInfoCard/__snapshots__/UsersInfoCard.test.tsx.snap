// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Users Info Card Component Test Users Info Card Component Snapshot 1`] = `
<DocumentFragment>
  <div
    class="grid-card"
  >
    <div
      class="card-icon align-items-center justify-content-center d-flex blue"
    >
      <div />
    </div>
    <div
      class="card-text"
    >
      hello world
    </div>
    <div
      class="card-text-bold"
    >
      hello world
    </div>
  </div>
</DocumentFragment>
`;
